source/validate.cpp

```diff
@@ -125,18 +125,24 @@ void DebugInstructionPass(ValidationState_t& _,
     default:
       break;
   }
-
 }
 
 // Collects use-def info about an instruction's IDs.
 void ProcessIds(ValidationState_t& _, const spv_parsed_instruction_t& inst) {
   if (inst.result_id) {
+    uint32_t curr_block = libspirv::kInvalidId;
+    if (_.get_current_function() &&
```

Comment:
The test against _.get_current_function() could be replaced with _.in_function_body()